### ISSUE TITLE
[ENHANCEMENT] Enhance the Renter API endpoints

### ISSUE THREAD

**Issue by mharkus** [NebulousLabs/Sia#2211]
Currently, the renter API uses a combination of POST and GET to perform CRUD operations. This can be improved by using other HTTP verbs to make the endpoints more concise. See the examples below.

### Upload a new file

Before: POST /renter/upload/*siapath
After:  POST /renter/*siapath

### Delete a file

Before: POST /renter/delete/*siapath
After: DELETE /renter/*siapath

### Rename a file

Before: POST /renter/rename/*siapath
After:  PUT /renter/*siapath

### Download a file

Before: GET /renter/download/*siapath
After:  GET /renter/*siapath

If you pass ?destination=path then it will become async download

### In summary, the new renter APIs will look like:

POST /renter/*siapath // upload
DELETE /renter/*siapath // delete
PUT /renter/*siapath // update
GET /renter/*siapath // retrieve

If this is ok, I will create a PR

**Comment by DavidVorick** [comment]
Worth mentioning, a primary reason we chose to support only GET and POST was because most libraries we found in various programming languages for interacting with http APIs only had strong native support for GET and POST, any other method often required additional work by the user.

We wanted to avoid this friction, and so made that sacrifice at the cost of being less restful.
